Modules/+Sources/Msquared.m

```diff
@@ -0,0 +1,481 @@
+                rethrow(err);
+            end
+        end
+        
+        function callGetWavelength(obj)             % Calls the solstis method 'get_wavelength' and fills prefs in.
+            if isempty(obj.hwserver) || ~isvalid(obj.hwserver) || isempty(obj.moduleName)
+                obj.status = obj.no_server;
+                obj.tuning = false;
+%                 obj.wavelength = NaN;
+                obj.wavelength_lock = NaN;
+            else
+                reply = obj.com('get_wavelength', 'solstis');
+                
+                obj.status = obj.statusList{reply.status+1};
+                obj.tuning = reply.status == 2;
+%                 obj.wavelength = reply.current_wavelength;    % We will preference the wavemeter.
+                obj.wavelength_lock = logical(reply.lock_status);
+            end
+        end
+        function callStatus(obj)                    % Calls the solstis method 'status' and fills prefs in.
+            % Get status report from SolsTiS laser and update fields
+            if isempty(obj.hwserver) || ~isvalid(obj.hwserver) || isempty(obj.moduleName)
+                obj.etalon_lock = NaN;
+                obj.etalon_voltage = NaN;
+%                 obj.resonator_lock = false;
+                obj.resonator_voltage = NaN;
+                obj.output_monitor = 0;
+            else
+                reply = obj.com('status', 'solstis');
+                
+                obj.etalon_lock =       strcmp(reply.etalon_lock,'on');
+                obj.etalon_voltage =    reply.etalon_voltage;
+%                 obj.resonator_lock =    strcmp(reply.cavity_lock,'on');
+                obj.resonator_voltage = reply.resonator_voltage;
+                obj.output_monitor =    reply.output_monitor;
+            end
+            
+            obj.callGetWavelength()
+        end
+        
+        function getWavemeterWavelength(obj)        % Measures the output wavelength directly from the wavemeter. Also uses the EMM channel if the EMM module is active.
+            if isempty(obj.hwserver) || ~isvalid(obj.hwserver)
+                obj.NIR_wavelength =  NaN;
+                obj.VIS_wavelength =  NaN;
+                obj.diff_wavelength = 1950;
+            else
+                obj.NIR_wavelength = obj.hwserver.com('wavemeter', 'GetWavelengthNum', obj.NIR_channel, 0);
+
+                if strcmp(obj.determineModule(obj.setpoint_), obj.moduleVIS)
+                    obj.hwserver.com('wavemeter', 'SetSwitcherSignalStates', obj.VIS_channel, 1, 1);
+                    obj.VIS_wavelength = obj.hwserver.com('wavemeter', 'GetWavelengthNum', obj.VIS_channel, 0);
+
+                    if obj.VIS_wavelength > 0 && obj.NIR_wavelength > 0
+                        diff_wavelength_ = 1/(1/obj.VIS_wavelength - 1/obj.NIR_wavelength);
+
+                        if diff_wavelength_ < 1951 && diff_wavelength_ > 1949   % This range is just so a silly value isn't set. Not sure if this is a good range.
```

Comment:
What should happen if the value is silly? Seems like at least an error/warning should be thrown

Reply:
In the middle of tuning, this value can be silly (maybe super multimode, bad timing, etc etc). By the time tuning is finished, there are other checks on silliness (checks on VIS and NIR wavelengths).